javascript/javascript3/week2/currencyCalculator.js

```diff
@@ -0,0 +1,151 @@
+
+//------------------------------------------------//
+
+// STEP 4: GET EXCHANGEINFO SAVED IN LOCAL STORAGE
+// Get exchangeInfo saved in Local Storage
+const exchangeInfoSaved = JSON.parse(localStorage.getItem("exchangeInfo"));
+
+if (exchangeInfoSaved) {
+    const countryCode = (exchangeInfoSaved["countryCode"])
+
+    for (let i = 0; i < countryCode.length; i++) {
+        // Create and append options for Country From
+        selectCountryFrom.innerHTML += `<option value="${countryCode[i]}">${countryCode[i]}</option>`
+        // Create and append options for Country To
+        selectCountryTo.innerHTML += `<option value="${countryCode[i]}">${countryCode[i]}</option>`
+    }
+}
+
+//-------------------------------------------------//
+
+// STEP 3: GET INPUT DATA FROM LOCAL STORAGE
+// Get Input Data saved in Local Storage
+const inputSaved = JSON.parse(localStorage.getItem("input"));
+
+// Check if input is saved in local storage.
+// If yes - display it as default.
+if (inputSaved) {
+    currencyFromInput.value = inputSaved["country-from"];
+    currencyToInput.value = inputSaved["country-to"];    
+    displayCurrencyFrom.innerHTML = `1 ${currencyFromInput.value} equals`;
+}
+//-------------------------------------------------//
+// STEP 1: FETCH DATA FROM API
+//Fetch Data from API
+const exchangeRatesData = async () => {
+
+    // When the convert button is clicked, calculate currency exchange.
+    let baseCurrency = currencyFromInput.value;
+   
+    // If location is blank, then default currency = "EUR"
+    if (baseCurrency == "") {
+        console.log("True");
+        baseCurrency = "EUR"
+    } else {
+        baseCurrency = currencyFromInput.value;        
+    }
+    
+    try {
+        const response = await fetch(`${BASE_URL}/v6/${API_KEY}/latest/${baseCurrency}`)
+        console.log(response.status);  //200 (Should be between 200 and 299)
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.log(error)
+    }
+
+};
+
+// STEP 2: SAVE DATA TO LOCAL STORAGE
+const saveToLocalStorage = async () => {
+    const data = await exchangeRatesData();
+
+    const exchangeData = await (data["conversion_rates"])
+    let countryCode = Object.keys(data["conversion_rates"])
+    
+    let rate = Object.values(data["conversion_rates"])
+   
+    const exchangeInfo = {
+        "countryCode": countryCode,
+        "rate": rate
+    }
+    
+    const input = {
+        "country-from": "EUR",
+        "country-to": "DKK"
+    }
+
+    // Save to Local Storage
+    localStorage.setItem("exchangeInfo", JSON.stringify(exchangeInfo))
+    localStorage.setItem("input", JSON.stringify(input))
+
+} // End of function saveToLocalStorage
+
+saveToLocalStorage()
+
+//------------------------------------------------//
+
+// Extract Data - To get exchange rates
+const display = async (event) => {
+    const data = await exchangeRatesData().then((data) => {
```

Comment:
You're combining promises and async await, which isn't quite right. In practice you should refrain from using the "then" keyword since that is the old way of doing things.

Reply:
Yes... Understood and removed .then